control_panel_api/views.py

```diff
@@ -22,28 +28,45 @@
 from control_panel_api.serializers import (
     AppS3BucketSerializer,
     AppSerializer,
-    UserAppSerializer,
     GroupSerializer,
     S3BucketSerializer,
+    UserAppSerializer,
     UserS3BucketSerializer,
     UserSerializer,
 )
 
+logger = logging.getLogger(__name__)
+
 
 class UserViewSet(viewsets.ModelViewSet):
     queryset = User.objects.all()
     serializer_class = UserSerializer
     filter_backends = (UserFilter,)
     permission_classes = (UserPermissions,)
 
+    @transaction.atomic
     def perform_create(self, serializer):
         instance = serializer.save()
-        instance.aws_create_role()
-        instance.helm_create()
 
+        try:
+            instance.aws_create_role()
+            instance.helm_create()
+        except ClientError as e:
+            logger.error(e)
+            raise AWSException
```

Comment:
This should be `raise AWSException(msg) from e` to chain the exceptions: https://docs.python.org/3/reference/simple_stmts.html#raise

Reply:
mindblown.jpg
👍

Reply:
@andyhd do you want to expose the full message "An error occurred (EntityAlreadyExists) when calling the CreateRole operation: Role with name foo already exists." in the output, or provide the generic "tehre was an AWS error.." and log?

Reply:
More detail is probably better